py/modmath.c

```diff
@@ -132,7 +132,24 @@ MATH_FUN_1(asin, asin)
 // atan(x)
 MATH_FUN_1(atan, atan)
 // atan2(y, x)
+#if defined(_MSC_VER) && (_MSC_VER <= 1800)
+// VC++ 12.0 and earlier return NaN if one of the arguments is Inf.
+mp_float_t msvc_atan2(mp_float_t x, mp_float_t y) {
+    if (isinf(x) && isinf(y)) {
+        static const mp_float_t pi_over_four = MICROPY_FLOAT_CONST(M_PI_4);
+        static const mp_float_t three_pi_over_four = MICROPY_FLOAT_CONST(3.0) * MICROPY_FLOAT_CONST(M_PI_4);
+        bool same_sign = signbit(x) == signbit(y);
+        if (x < MICROPY_FLOAT_CONST(0.0)) {
+            return -(same_sign ? three_pi_over_four : pi_over_four);
+        }
+        return same_sign ? pi_over_four : three_pi_over_four;
```

Comment:
Is it simpler to write it like: `sign(x) * (y < 0 ? three_pi_over_four : pi_over_four)`?

Reply:
Even better: `copysign(y < 0 ? three_pi_over_four : pi_over_four, x)`

Reply:
Nice, didn't think of copysign.